x-pack/server/lib/register_route/index.js

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { callWithRequestFactory } from './call_with_request_factory';
+import { isEsErrorFactory } from './is_es_error_factory';
+import { wrapEsError, wrapUnknownError } from './error_wrappers';
+import { licensePreRoutingFactory } from'./license_pre_routing_factory';
+
+export function registerRoute({ server, handler, pluginId, path, method }) {
+  const isEsError = isEsErrorFactory(server);
+  const licensePreRouting = licensePreRoutingFactory(server, pluginId);
+
+  server.route({
+    path,
+    method,
+    handler: async (request, h) => {
+      const callWithRequest = callWithRequestFactory(server, request);
+      try {
```

Comment:
I'm concerned about the inflexibility of hardcoding our error-handling this way. There are edge cases to the way we handle our errors, e.g. [rollups](https://github.com/elastic/kibana/blob/master/x-pack/plugins/rollup/server/routes/api/indices.js#L101), so I'd find value in a solution that makes it easier to handle the default case, but still allows for custom logic. For example, something like this:

```js

const handler = async ({ request, callWithRequest, h, handleDefaultError }) => {
  try {
    const { indices = [] } = request.payload;
    const params = {
      path: `/${encodeURIComponent(indices.join(','))}/_freeze`,
      method: 'POST',
    };

    await callWithRequest('transport.request', params);
    return h.response();
  } catch (err) {
    throw handleDefaultError(err);
  }  
};
```

Reply:
I just realized that the existing proposal supports this already. The consumer just needs to put a `try/catch` in the handler to handle the edge-case, and re-throw the error if the edge-case isn't met. So you can ignore my comment above. 😄